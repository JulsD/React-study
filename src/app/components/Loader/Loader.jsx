import React from 'react';
import styles from './loader.css';

class Loader extends React.Component {

  render() {
    return (
      <div className = {styles.root}></div>
    );
  }

}

export default Loader;
