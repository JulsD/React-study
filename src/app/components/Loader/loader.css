.root {
  margin: 0 auto;
  border: 10px solid #666;
  border-top: 10px solid #f40612;
  border-radius: 50%;
  width: 70px;
  height: 70px;
  animation: spin 2s ease-out infinite;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}