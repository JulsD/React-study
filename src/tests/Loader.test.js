import React from 'react';
import Loader from '../app/components/Loader';

test('Loader should render a <div>', () => {
  const wrapper = shallow(
    <Loader />
  );
  expect(wrapper).toMatchSnapshot();
});
