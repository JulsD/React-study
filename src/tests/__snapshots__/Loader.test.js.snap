// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render a <div> 1`] = `
<div
  className="root"
/>
`;
